diff --git a/Makefile.am b/Makefile.am
index cc3024e68..e58f8688c 100644
--- a/Makefile.am	
+++ b/Makefile.am
@@ -24,7 +24,7 @@
 SUBDIRS=fontconfig fc-case fc-lang src \
 	fc-cache fc-cat fc-conflist fc-list fc-match \
 	fc-pattern fc-query fc-scan fc-validate conf.d \
-	its po po-conf test
+	its po po-conf
 if ENABLE_DOCS
 SUBDIRS += doc
 endif
diff --git a/conf.d/Makefile.am b/conf.d/Makefile.am
index 710318c14..c404666e0 100644
--- a/conf.d/Makefile.am	
+++ b/conf.d/Makefile.am
@@ -100,6 +100,9 @@ README: $(srcdir)/README.in
 35-lang-normalize.conf: ../fc-lang/Makefile.am
 	cd ../fc-lang && $(MAKE) $(AM_MAKEFLAGS) $(top_builddir)/conf.d/35-lang-normalize.conf
 
+if OS_WIN32
+install-data-hook:
+else
 install-data-hook:
 	mkdir -p $(DESTDIR)$(configdir)
 	@(echo cd $(DESTDIR)$(configdir);			\
@@ -116,5 +119,5 @@ uninstall-local:
 	    echo $(RM) $$i;					\
 	    $(RM) $$i;						\
 	  done)
-
+endif
 -include $(top_srcdir)/git.mk
diff --git a/fc-cache/fc-cache.c b/fc-cache/fc-cache.c
index 091eed494..4138d0b16 100644
--- a/fc-cache/fc-cache.c	
+++ b/fc-cache/fc-cache.c
@@ -34,7 +34,9 @@
 #include <fontconfig/fontconfig.h>
 #include <stdio.h>
 #include <stdlib.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <errno.h>
diff --git a/fc-cat/fc-cat.c b/fc-cat/fc-cat.c
index 69611bcc3..e2be0cb7f 100644
--- a/fc-cat/fc-cat.c	
+++ b/fc-cat/fc-cat.c
@@ -36,7 +36,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <errno.h>
new file mode 100644
index 000000000..3b2dba36b
--- /dev/null
+++ b/fc-conflist/.gitignore
@@ -0,0 +1,24 @@
+/*.bak
+/*.lo
+/*.obj
+/*.orig
+/*.rej
+/*.tab.c
+/*~
+/.*.sw[nop]
+/.deps
+/.dirstamp
+/.gitignore
+/.libs
+/GPATH
+/GRTAGS
+/GSYMS
+/GTAGS
+/ID
+/Makefile
+/Makefile.in
+/TAGS
+/_libs
+/fc-conflist.exe
+/so_locations
+/tags
diff --git a/fc-conflist/fc-conflist.c b/fc-conflist/fc-conflist.c
index 5c40a0f09..74d2d9c7e 100644
--- a/fc-conflist/fc-conflist.c	
+++ b/fc-conflist/fc-conflist.c
@@ -35,7 +35,9 @@
 
 #include <fontconfig/fontconfig.h>
 #include <stdio.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 #include <locale.h>
diff --git a/fc-list/fc-list.c b/fc-list/fc-list.c
index 2039acd36..6fc829bf9 100644
--- a/fc-list/fc-list.c	
+++ b/fc-list/fc-list.c
@@ -24,7 +24,9 @@
 
 #include <fontconfig/fontconfig.h>
 #include <stdio.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 #include <locale.h>
diff --git a/fc-match/fc-match.c b/fc-match/fc-match.c
index dee61479d..8ff972160 100644
--- a/fc-match/fc-match.c	
+++ b/fc-match/fc-match.c
@@ -33,7 +33,9 @@
 
 #include <fontconfig/fontconfig.h>
 #include <stdio.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 #include <locale.h>
diff --git a/fc-pattern/fc-pattern.c b/fc-pattern/fc-pattern.c
index 7989b816f..c46b6c021 100644
--- a/fc-pattern/fc-pattern.c	
+++ b/fc-pattern/fc-pattern.c
@@ -33,7 +33,9 @@
 
 #include <fontconfig/fontconfig.h>
 #include <stdio.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 #include <locale.h>
diff --git a/fc-query/fc-query.c b/fc-query/fc-query.c
index fbffb8415..94024c671 100644
--- a/fc-query/fc-query.c	
+++ b/fc-query/fc-query.c
@@ -36,7 +36,9 @@
 #include <fontconfig/fontconfig.h>
 #include <fontconfig/fcfreetype.h>
 #include <stdio.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 #include <locale.h>
diff --git a/fc-scan/fc-scan.c b/fc-scan/fc-scan.c
index 9302ac559..acd3ab29d 100644
--- a/fc-scan/fc-scan.c	
+++ b/fc-scan/fc-scan.c
@@ -36,7 +36,9 @@
 #include <fontconfig/fontconfig.h>
 #include <fontconfig/fcfreetype.h>
 #include <stdio.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 #include <locale.h>
diff --git a/fc-validate/fc-validate.c b/fc-validate/fc-validate.c
index 327ae63fb..48752de33 100644
--- a/fc-validate/fc-validate.c	
+++ b/fc-validate/fc-validate.c
@@ -36,7 +36,9 @@
 #include <fontconfig/fontconfig.h>
 #include <fontconfig/fcfreetype.h>
 #include <stdio.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 #include <locale.h>
diff --git a/src/Makefile.am b/src/Makefile.am
index 35e820d60..42bd7c5a2 100644
--- a/src/Makefile.am	
+++ b/src/Makefile.am
@@ -33,7 +33,7 @@ fontconfig_def_dependency = fontconfig.def
 
 install-libtool-import-lib: libfontconfig.la
 	$(MKDIR_P) $(DESTDIR)$(libdir)
-	$(INSTALL) .libs/libfontconfig.dll.a $(DESTDIR)$(libdir)/libfontconfig.dll.a
+#	$(INSTALL) .libs/libfontconfig.dll.a $(DESTDIR)$(libdir)/libfontconfig.dll.a
 	$(INSTALL) fontconfig.def $(DESTDIR)$(libdir)/fontconfig.def
 
 uninstall-libtool-import-lib:
@@ -58,7 +58,8 @@ fontconfig.lib : libfontconfig.la
 	lib -name:libfontconfig-@LIBT_CURRENT_MINUS_AGE@.dll -def:fontconfig.def -out:$@
 
 install-ms-import-lib:
-	$(INSTALL) fontconfig.lib $(DESTDIR)$(libdir)
+	$(MKDIR_P) $(DESTDIR)$(libdir)
+	$(INSTALL) .libs/fontconfig.lib $(DESTDIR)$(libdir)
 
 uninstall-ms-import-lib:
 	$(RM) $(DESTDIR)$(libdir)/fontconfig.lib
diff --git a/src/fcatomic.c b/src/fcatomic.c
index d12d32408..da7234934 100644
--- a/src/fcatomic.c	
+++ b/src/fcatomic.c
@@ -50,7 +50,9 @@
 #include "fcint.h"
 #include <sys/types.h>
 #include <sys/stat.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <stdlib.h>
 #include <time.h>
 
diff --git a/src/fccache.c b/src/fccache.c
index 097620141..f137f749d 100644
--- a/src/fccache.c	
+++ b/src/fccache.c
@@ -30,9 +30,9 @@
 #include <limits.h>
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <sys/time.h>
+//#include <sys/time.h>
 #include <assert.h>
-#if defined(HAVE_MMAP) || defined(__CYGWIN__)
+#if defined(HAVE_MMAP) || defined(__CYGWIN__) || defined(HAVE_UNISTD_H)
 #  include <unistd.h>
 #  include <sys/mman.h>
 #endif
diff --git a/src/fccompat.c b/src/fccompat.c
index 16f108d10..b8b5bb7d7 100644
--- a/src/fccompat.c	
+++ b/src/fccompat.c
@@ -34,6 +34,10 @@
 #if HAVE_SYS_STAT_H
 #include <sys/stat.h>
 #endif
+#ifdef _WIN32
+#include <direct.h>
+#define mkdir(path,mode) _mkdir(path)
+#endif
 #if HAVE_FCNTL_H
 #include <fcntl.h>
 #endif
@@ -232,11 +236,6 @@ FcRandom(void)
     return result;
 }
 
-#ifdef _WIN32
-#include <direct.h>
-#define mkdir(path,mode) _mkdir(path)
-#endif
-
 FcBool
 FcMakeDirectory (const FcChar8 *dir)
 {
diff --git a/src/fcint.h b/src/fcint.h
index 9e1da1ee5..d2278871a 100644
--- a/src/fcint.h	
+++ b/src/fcint.h
@@ -40,10 +40,33 @@
 #include <limits.h>
 #include <float.h>
 #include <math.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#else
+    
+#ifdef _MSC_VER
+#include <BaseTsd.h>
+#include <io.h>
+typedef SSIZE_T ssize_t;
+typedef int mode_t;
+#ifndef R_OK
+#define R_OK 4
+#endif
+#ifndef W_OK
+#define W_OK 2
+#endif
+#ifndef F_OK
+#define F_OK 0
+#endif
+#endif
+
+#endif
 #include <stddef.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#ifdef _MSC_VER
+#include <direct.h>
+#endif
 #include <time.h>
 #include <fontconfig/fontconfig.h>
 #include <fontconfig/fcprivate.h>
diff --git a/src/fcstr.c b/src/fcstr.c
index cdebe983b..eaf793ea8 100644
--- a/src/fcstr.c	
+++ b/src/fcstr.c
@@ -26,7 +26,9 @@
 #include <stdlib.h>
 #include <ctype.h>
 #include <string.h>
-
+#ifdef _MSC_VER
+#include <direct.h>
+#endif
 
 /* Objects MT-safe for readonly access. */
 
@@ -1134,7 +1136,7 @@ FcConvertDosPath (char *str)
 FcChar8 *
 FcStrCanonFilename (const FcChar8 *s)
 {
-#ifdef _WIN32
+#if defined(_WIN32) || defined(_MSC_VER)
     FcChar8 full[FC_MAX_FILE_LEN + 2];
     int size = GetFullPathName ((LPCSTR) s, sizeof (full) -1,
 				(LPSTR) full, NULL);
diff --git a/test/test-bz106632.c b/test/test-bz106632.c
index 5db912b12..fd29a63bc 100644
--- a/test/test-bz106632.c	
+++ b/test/test-bz106632.c
@@ -29,7 +29,9 @@
 #include <stdlib.h>
 #include <string.h>
 #include <dirent.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <errno.h>
 #ifndef HAVE_STRUCT_DIRENT_D_TYPE
 #include <sys/types.h>
diff --git a/test/test-d1f48f11.c b/test/test-d1f48f11.c
index 929afd932..76866177a 100644
--- a/test/test-d1f48f11.c	
+++ b/test/test-d1f48f11.c
@@ -29,7 +29,9 @@
 #include <stdlib.h>
 #include <string.h>
 #include <dirent.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <errno.h>
 #ifndef HAVE_STRUCT_DIRENT_D_TYPE
 #include <sys/types.h>
diff --git a/test/test-issue107.c b/test/test-issue107.c
index bc2ee25e7..7120f0ee3 100644
--- a/test/test-issue107.c	
+++ b/test/test-issue107.c
@@ -29,7 +29,28 @@
 #include <stdlib.h>
 #include <string.h>
 #include <dirent.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#else
+    
+#ifdef _MSC_VER
+#include <Windows.h>
+#include <BaseTsd.h>
+#include <io.h>
+typedef SSIZE_T ssize_t;
+typedef int mode_t;
+#ifndef R_OK
+#define R_OK 4
+#endif
+#ifndef W_OK
+#define W_OK 2
+#endif
+#ifndef F_OK
+#define F_OK 0
+#endif
+#endif
+
+#endif
 #include <errno.h>
 #ifndef HAVE_STRUCT_DIRENT_D_TYPE
 #include <sys/types.h>
diff --git a/test/test-issue110.c b/test/test-issue110.c
index 3794b08e5..a6cf5900f 100644
--- a/test/test-issue110.c	
+++ b/test/test-issue110.c
@@ -29,7 +29,9 @@
 #include <stdlib.h>
 #include <string.h>
 #include <dirent.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <errno.h>
 #ifndef HAVE_STRUCT_DIRENT_D_TYPE
 #include <sys/types.h>
diff --git a/test/test-migration.c b/test/test-migration.c
index b7d96dcce..5f42eab2e 100644
--- a/test/test-migration.c	
+++ b/test/test-migration.c
@@ -28,13 +28,14 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <sys/types.h>
 #include <dirent.h>
 #ifndef HAVE_STRUCT_DIRENT_D_TYPE
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <unistd.h>
 #endif
 #include <fontconfig/fontconfig.h>
 
diff --git a/test/test-pthread.c b/test/test-pthread.c
index fbf397d4c..6c3bea396 100644
--- a/test/test-pthread.c	
+++ b/test/test-pthread.c
@@ -24,7 +24,9 @@
  */
 #include <stdio.h>
 #include <stdlib.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <pthread.h>
 #include <fontconfig/fontconfig.h>
 
