diff --git a/qtwebengine/src/3rdparty/chromium/tools/gn/bootstrap/bootstrap.py b/qtwebengine/src/3rdparty/chromium/tools/gn/bootstrap/bootstrap.py
index a5b69a2..01c5cde 100644
--- a/qtwebengine/src/3rdparty/chromium/tools/gn/bootstrap/bootstrap.py
+++ b/qtwebengine/src/3rdparty/chromium/tools/gn/bootstrap/bootstrap.py
@@ -159,7 +159,12 @@
                 ]
             },
             "sources": [
-                "-lpcre2-16"
+                {
+                    "builds": {
+                        "debug": "-lpcre2-16d",
+                        "release": "-lpcre2-16"
+                    }
+                }
             ]
         },
         "pps": {

C:\Sources\Extern\vcpkg\buildtrees\qt5\src\qt-5.9.3\qtwebengine/src/3rdparty/chromium/tools/gn/bootstrap/bootstrap.py