diff --git a/qtwebengine/src/3rdparty/chromium/build/toolchain/get_concurrent_links.py b/qtwebengine/src/3rdparty/chromium/build/toolchain/get_concurrent_links.py
index a5b69a2..01c5cde 100644
--- a/qtwebengine/src/3rdparty/chromium/build/toolchain/get_concurrent_links.py
+++ b/qtwebengine/src/3rdparty/chromium/build/toolchain/get_concurrent_links.py
@@ -70,5 +70,4 @@
   options, _ = parser.parse_args()
 
-  print _GetDefaultConcurrentLinks(options.mem_per_link_gb,
-                                    options.reserve_mem_gb)
+  print _GetDefaultConcurrentLinks(options.mem_per_link_gb, options.reserve_mem_gb)
   return 0
diff --git a/qtwebengine/src/3rdparty/chromium/mojo/public/tools/bindings/mojom_bindings_generator.py b/qtwebengine/src/3rdparty/chromium/mojo/public/tools/bindings/mojom_bindings_generator.py
index a5b69a2..01c5cde 100644
--- a/qtwebengine/src/3rdparty/chromium/mojo/public/tools/bindings/mojom_bindings_generator.py
+++ b/qtwebengine/src/3rdparty/chromium/mojo/public/tools/bindings/mojom_bindings_generator.py
@@ -26,12 +26,12 @@
     if rel_filename.path in imported_filename_stack:
-      print "%s: Error: Circular dependency" % rel_filename.path + \
+      print("%s: Error: Circular dependency" % rel_filename.path + \
-          MakeImportStackMessage(imported_filename_stack + [rel_filename.path])
+          MakeImportStackMessage(imported_filename_stack + [rel_filename.path]))
       sys.exit(1)
 
     try:
       with open(rel_filename.path) as f:
         source = f.read()
     except IOError as e:
-      print "%s: Error: %s" % (e.rel_filename.path, e.strerror) + \
-          MakeImportStackMessage(imported_filename_stack + [rel_filename.path])
+      print("%s: Error: %s" % (rel_filename.path, e.strerror) + \
+          MakeImportStackMessage(imported_filename_stack + [rel_filename.path]))
       sys.exit(1)
