diff --git a/qtwebengine/src/3rdparty/chromium/tools/gn/bootstrap/bootstrap.py b/qtwebengine/src/3rdparty/chromium/tools/gn/bootstrap/bootstrap.py
index a5b69a2..01c5cde 100644
--- a/qtwebengine/src/3rdparty/chromium/tools/gn/bootstrap/bootstrap.py
+++ b/qtwebengine/src/3rdparty/chromium/tools/gn/bootstrap/bootstrap.py
@@ -70,4 +70,4 @@
     build_root = os.getcwd()
 
-  print 'Building gn manually in a temporary directory for bootstrapping...' 
+  print('Building gn manually in a temporary directory for bootstrapping...')
   build_gn_with_ninja_manually(tempdir, options)
@@ -85,3 +85,3 @@
   else:
-    print 'Building gn using itself to %s...' % build_rel
+    print('Building gn using itself to %s...' % build_rel)
     build_gn_with_gn(temp_gn, build_root, options)
@@ -262,4 +262,4 @@	 
     ])
 
-  for library, settings in static_libraries.iteritems():
+  for library, settings in static_libraries.items():
     for src_file in settings['sources']:
@@ -270,4 +270,4 @@	 
         ' '.join([src_to_obj(src_file) for src_file in settings['sources']])))
 
-  for executable, settings in executables.iteritems():
+  for executable, settings in executables.items():
     for src_file in settings['sources']:	 
@@ -129,4 +129,4 @@
         return run_build(tempdir, options)
   except subprocess.CalledProcessError as e:
-    print >> sys.stderr, str(e)
+    sys.stderr.write(str(e))
     return 1	 
diff --git a/qtwebengine/src/3rdparty/chromium/build/toolchain/win/setup_toolchain.py b/qtwebengine/src/3rdparty/chromium/build/toolchain/win/setup_toolchain.py
index a5b69a2..01c5cde 100644
--- a/qtwebengine/src/3rdparty/chromium/build/toolchain/win/setup_toolchain.py
+++ b/qtwebengine/src/3rdparty/chromium/build/toolchain/win/setup_toolchain.py
@@ -208,10 +208,10 @@
     with open('environment.winrt_' + cpu, 'wb') as f:
       f.write(env_block)
 
   assert vc_bin_dir
-  print 'vc_bin_dir = ' + gn_helpers.ToGNString(vc_bin_dir)
+  print('vc_bin_dir = ' + gn_helpers.ToGNString(vc_bin_dir))
   assert include
-  print 'include_flags = ' + gn_helpers.ToGNString(include)
+  print('include_flags = ' + gn_helpers.ToGNString(include))
 
 if __name__ == '__main__':
   main()
diff --git a/qtwebengine/src/3rdparty/chromium/build/win_is_xtree_patched.py b/qtwebengine/src/3rdparty/chromium/build/win_is_xtree_patched.py
index a5b69a2..01c5cde 100644
--- a/qtwebengine/src/3rdparty/chromium/build/win_is_xtree_patched.py
+++ b/qtwebengine/src/3rdparty/chromium/build/win_is_xtree_patched.py
@@ -26,1 +26,1 @@
-    print DoMain([])
+    print(DoMain([]))
diff --git a/qtwebengine/src/3rdparty/chromium/google_apis/build/check_internal.py b/qtwebengine/src/3rdparty/chromium/google_apis/build/check_internal.py
index a5b69a2..01c5cde 100644
--- a/qtwebengine/src/3rdparty/chromium/google_apis/build/check_internal.py
+++ b/qtwebengine/src/3rdparty/chromium/google_apis/build/check_internal.py
@@ -17,4 +17,4 @@
   if os.path.exists(sys.argv[1]):
-    print 1
+    print(1)
   else:
-    print 0
+    print(0)
diff --git a/qtwebengine/src/3rdparty/chromium/build/gypi_to_gn.py b/qtwebengine/src/3rdparty/chromium/build/gypi_to_gn.py
index a5b69a2..01c5cde 100644
--- a/qtwebengine/src/3rdparty/chromium/build/gypi_to_gn.py
+++ b/qtwebengine/src/3rdparty/chromium/build/gypi_to_gn.py
@@ -103,5 +103,5 @@
-  except SyntaxError, e:
+  except SyntaxError as e:
     e.filename = path
     raise
-  except Exception, e:
+  except Exception as e:
     raise Exception("Unexpected error while reading %s: %s" % (path, str(e)))
@@ -184,1 +184,1 @@	 
-  print gn_helpers.ToGNString(data)
+  print(gn_helpers.ToGNString(data))
@@ -186,5 +186,5 @@	 
 if __name__ == '__main__':
   try:
     main()
-  except Exception, e:
+  except Exception as e:
-    print str(e)
+    print(str(e))
     sys.exit(1)
diff --git a/qtwebengine/src/3rdparty/chromium/build/util/version.py b/qtwebengine/src/3rdparty/chromium/build/util/version.py
index a5b69a2..01c5cde 100644
--- a/qtwebengine/src/3rdparty/chromium/build/util/version.py
+++ b/qtwebengine/src/3rdparty/chromium/build/util/version.py
@@ -67,1 +67,1 @@
-  for key, val in values.iteritems():
+  for key, val in values.tems():	
@@ -71,1 +71,1 @@
-      print repr(key), repr(val)
+      print(repr(key), repr(val))
@@ -164,1 +164,1 @@
-    print contents
+    print(contents)