diff --git a/configure.ac b/configure.ac
index ae619d446..20be62406 100644
--- a/configure.ac	
+++ b/configure.ac
@@ -19,9 +19,13 @@ AC_PREREQ([2.63])
 AC_INIT([libiconv], [1.16])
 AC_CONFIG_SRCDIR([lib/iconv.c])
 AC_CONFIG_AUX_DIR([build-aux])
+AC_CONFIG_MACRO_DIRS([m4])
+AC_CONFIG_MACRO_DIRS([srcm4])
+m4_pattern_allow([AC_EILSEQ])
 AM_INIT_AUTOMAKE
 AC_CONFIG_HEADERS([config.h lib/config.h])
 AC_PROG_MAKE_SET
+AC_USE_SYSTEM_EXTENSIONS
 
 dnl           checks for basic programs
 
